config/src/main/java/tech/pegasys/pantheon/config/CliqueConfigOptions.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2018 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.config;
+
+import io.vertx.core.json.JsonObject;
+
+public class CliqueConfigOptions {
+
+  public static final CliqueConfigOptions DEFAULT = new CliqueConfigOptions(new JsonObject());
+
+  private static final long DEFAULT_EPOCH_LENGTH = 30_000;
+  private static final int DEFAULT_BLOCK_PERIOD_SECONDS = 15;
+
+  private final JsonObject cliqueConfigRoot;
+
+  public CliqueConfigOptions(final JsonObject cliqueConfigRoot) {
+    this.cliqueConfigRoot = cliqueConfigRoot;
```

Comment:
nit/ymmv: is it worth pre-parsing the config such that if its bad, we fail at startup rather than when we go to use the config?

Reply:
I don't think these config objects have the primary responsibility for validation - they do some incidental validation in terms of "must be a number" but even then a lot of the time they just return a `String` and leave parsing to the caller.

The values from these config objects are actually used during startup anyway and it would be a smell to me if the config objects continued to live on past startup.

Reply:
Right now, the same place it currently does - at the point of usage.  I can see some benefit in having specific validation for config objects but I think that would be a separate validator object that uses these config objects to extract the data to validate. I think this is a big enough PR without attempting to rework validation at the same time.

Reply:
Happy to revisit it later in another PR. My thoughts to carry forward are I think the validation should come earlier than it's point of usage so whatever is being used can be treated as untainted.